rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSourceDescriptor.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.resourcemanager.nodesource;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.ow2.proactive.resourcemanager.authentication.Client;
+
+
+/**
+ *
+ */
+public class NodeSourceDescriptor {
+
+    private String name;
+
+    private String infrastructureType;
+
+    private Object[] infrastructureParameters;
+
+    private String policyType;
+
+    private Object[] policyParameters;
+
+    private Client provider;
+
+    private String description;
+
+    private boolean nodesRecoverable;
+
+    private NodeSourceStatus status;
+
+    private Map<String, Serializable> lastRecoveredInfrastructureVariables;
+
+    public NodeSourceDescriptor(String name, String infrastructureType, Object[] infrastructureParameters,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 10 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)